### DISCUSSION TITLE
Can a form auto-focus the first invalid field after failed submission?

### DISCUSSION THREAD

**Discussion by gr-benjamin-delong** [w3c/wcag#3414] [Q&A] [answered]
I note that a submit button can initiate a change of context (e.g. change of focus): https://www.w3.org/WAI/WCAG21/Techniques/general/G80

> this is an appropriate control to use for causing a change of context and is a practice that does not create confusion for users.

Would that include auto-focusing the first invalid field?

**Comment by patrickhlauke** [comment, marked as answer]
I would say yes ... the user is expecting a change of context when they hit submit. So it's not surprising to them. Yes, they may expect it to go to a "thank you, your submission happened" confirmation, but if they get bounced back to the first field that's in error, that's fine.
